src/pyoncatqt/login.py

```diff
@@ -215,11 +221,15 @@ def __init__(self: QGroupBox, key: str = None, parent: QWidget = None, **kwargs:
         # OnCat agent
 
         self.oncat_url = get_data("login.oncat", "oncat_url")
-        self.client_id = get_data("login.oncat", f"{key}_id")
+        self.client_id = client_id or get_data("login.oncat", f"{key}_id")
```

Comment:
Small problem here. If neither is defined then the second half is `get_data("login.oncat", "None_id")` which gives the right behavior but seems a little dodgy. It may be better to expand this out.
```suggestion
        if (client_id is not None):
            self.client_id = client_id
        else if (key is not None):
            self.client_id = get_data("login.oncat", f"{key}_id")
        else:
            raise ValueError(f"Invalid module {key}. No OnCat client Id is found or provided for this application.")
```

I've not seen this syntax before. Thanks for the pointer.

Reply:
it is another way to write the above in fewer lines. I'll update the code, if it creates issues. I added it, because it looked cleaner to me.